### DISCUSSION TITLE
How to backup single-node docker couchDB by storing data dir?

### DISCUSSION THREAD

**Discussion by irinhwng** [apache/couchdb#4470] [Q&A] [answered]
According to documentation, I can save .couch, .shards, and shard files which all reside in /data. To test out whether all the data can be saved, I have two single-node couchDBs in two separate namespaces on openshift. For simplicity, they're called couchdb-pod1 and couchdb-pod2. Note: couchdb-pod1 has all of my dbs while couchdb-pod2 is completely empty.

Here's what I've done:

1. copy everything in /data and store it on my local machine

```
oc cp couchdb-pod1:/opt/couchdb/data $HOME/Desktop/data
```

2. once loaded, replace empty /data from couchdb-pod2 with data from step 1
```
oc cp /Users/me/Desktop/data couchdb-pod2:/opt/couchdb/data
```

Here's the issue: My original couch has x amount of records for each DB (see pic)
<img width="866" alt="pod1" src="https://user-images.githubusercontent.com/49319582/224407135-782936d4-fb49-4450-b353-7ebc3a14aca9.png">

Once I load the data into couchdb-pod2, I'm getting the majority of the data but not all of it.
<img width="946" alt="Screen Shot 2023-03-03 at 3 59 15 PM" src="https://user-images.githubusercontent.com/49319582/224407203-968c149a-b409-4051-9ebf-67ac3b707e36.png">

What am I doing wrong here? I don't need to touch the vm.arg file since this is single node (thats my assumption so please correct me if I'm wrong).

**Comment by big-r81** [comment, marked as answer]
Hi,

1. Do your pods have exact the same (CouchDB-) settings?
2. Can you check, if your copied data (after backup) has the same size of your source (pod1)?
3. Is there anything in the logs of pod2?

**Reply by irinhwng** [reply]
Thanks for the quick response. Here are my answers to your questions:
1. yes - both pods have the same settings
2. copied data size: ~528 MBs whereas the data dir in pod has ~469MBs (yes, copied data is bigger than pod data which I thought was interesting)
3. found of few `notice` messages but the main error I got was the following:

```
[error] 2023-03-10T21:07:31.568681Z nonode@nohost emulator -------- Error in process <0.19916.152> with exit value:
{database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,400}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,375}]},{mem3_shards,load_shards_from_disk,2,[{file,"src/mem3_shards.erl"},{line,404}]},{mem3_shards,for_docid,3,[{file,"src/mem3_shards.erl"},{line,97}]},{fabric_doc_open,go,3,[{file,"src/fabric_doc_open.erl"},{line,39}]},{chttpd_auth_cache,ensure_auth_ddoc_exists,2,[{file,"src/chttpd_auth_cache.erl"},{line,198}]},{chttpd_auth_cache,listen_for_changes,1,[{file,"src/chttpd_auth_cache.erl"},{line,145}]}]}
```


These questions were really helpful since they got me digging into pod2. I'm going to run this again and double-check that I don't have duplicate files in the copied directory. When I rsh'ed into pod2, I noticed the `/data` dir was nested (`/opt/couchdb/data/data/data`). 

I'm having operation permissions with the cluster, so stay tuned.

**Reply by irinhwng** [reply]
*operation permission ISSUES

**Reply by irinhwng** [reply]
update: data dir was heavily nested - I tried it again with root access and every record was restored :)
